py4DSTEM/io/nonnative/read_dm.py

```diff
@@ -41,35 +41,37 @@ def read_dm(filepath, name="dm_dataset", mem="RAM", binfactor=1, **kwargs):
                 dataset_index = i
                 break
 
-        # The pixel sizes of all datasets are chained together, so
-        # we have to figure out the right offset
-        scale_offset = sum(dmFile.dataShape[:dataset_index]) + 2 * thumbanil_count
-        pixelsize = dmFile.scale[scale_offset:]
-        pixelunits = dmFile.scaleUnit[scale_offset:]
-
-        # Get the calibration pixel sizes
-        Q_pixel_size = pixelsize[0]
-        Q_pixel_units = "pixels" if pixelunits[0] == "" else pixelunits[0]
-        R_pixel_size = pixelsize[2]
-        R_pixel_units = "pixels" if pixelunits[2] == "" else pixelunits[2]
-
-        # Check that the units are sensible
-        # On microscopes that do not have live communication with the detector
-        # the calibrations can be invalid
-        if Q_pixel_units in ("nm", "µm"):
-            Q_pixel_units = "pixels"
-            Q_pixel_size = 1
-            R_pixel_units = "pixels"
-            R_pixel_size = 1
-
-        # Convert mrad to Å^-1 if possible
-        if Q_pixel_units == "mrad":
-            voltage = [v for t,v in dmFile.allTags.items() if "Microscope Info.Voltage" in t]
-            if len(voltage) == 1:
-                from py4DSTEM.process.utils import electron_wavelength_angstrom
-                lamda = electron_wavelength_angstrom(voltage[0])
-                Q_pixel_units = "A^-1"
-                Q_pixel_size = Q_pixel_size / lamda / 1000. # convert mrad to 1/Å
+        # We will only try to read pixel sizes for 4D data for now
+        if dmFile.dataShape[dataset_index] > 2:
```

Comment:
will this not break for 3D stacks? perhaps this should be `== 4`?

Reply:
Titan data is 3-D inside the file, so we let it pass here. If the special TitanX tags are found it reshapes the data to 4D, and then at a later step we enforce `==4`

Reply:
ah gotcha, the `try...except` statements take care of it